src/test/java/com/google/devtools/build/lib/analysis/RunfilesRepoMappingManifestTest.java

```diff
@@ -0,0 +1,274 @@
+  private Path moduleRoot;
+  private FakeRegistry registry;
+
+  @Override
+  protected ImmutableList<Injected> extraPrecomputedValues() {
+    try {
+      moduleRoot = scratch.dir("modules");
+    } catch (IOException e) {
+      throw new IllegalStateException(e);
+    }
+    registry = FakeRegistry.DEFAULT_FACTORY.newFakeRegistry(moduleRoot.getPathString());
+    return ImmutableList.of(
+        PrecomputedValue.injected(
+            ModuleFileFunction.REGISTRIES, ImmutableList.of(registry.getUrl())),
+        PrecomputedValue.injected(ModuleFileFunction.IGNORE_DEV_DEPS, false),
+        PrecomputedValue.injected(ModuleFileFunction.MODULE_OVERRIDES, ImmutableMap.of()),
+        PrecomputedValue.injected(
+            BazelModuleResolutionFunction.CHECK_DIRECT_DEPENDENCIES, CheckDirectDepsMode.WARNING));
+  }
+
+  @Override
+  protected AnalysisMock getAnalysisMock() {
+    // Make sure we don't have built-in modules affecting the dependency graph.
+    return new AnalysisMock.Delegate(super.getAnalysisMock()) {
+      @Override
+      public ImmutableMap<SkyFunctionName, SkyFunction> getSkyFunctions(
+          BlazeDirectories directories) {
+        return ImmutableMap.<SkyFunctionName, SkyFunction>builder()
+            .putAll(
+                Maps.filterKeys(
+                    super.getSkyFunctions(directories),
+                    fnName -> !fnName.equals(SkyFunctions.MODULE_FILE)))
+            .put(
+                SkyFunctions.MODULE_FILE,
+                new ModuleFileFunction(
+                    FakeRegistry.DEFAULT_FACTORY, directories.getWorkspace(), ImmutableMap.of()))
+            .buildOrThrow();
+      }
+    };
+  }
+
+  /**
+   * Sets up a Bazel module simple_rule@1.0, which provides a simple_binary rule that passes along
+   * runfiles in the data attribute. It also uses a toolchain (type //:toolchain_type, default
+   * toolchain //:simple_toolchain; the macro //:defs.bzl%simple_toolchain can be used to create
+   * more).
+   */
+  @Before
+  public void setupSimpleBinaryRule() throws Exception {
+    setBuildLanguageOptions("--enable_bzlmod");
+    registry.addModule(
+        createModuleKey("simple_rule", "1.0"),
+        "module(name='simple_rule',version='1.0')",
+        "register_toolchains('//:all')");
+    scratch.overwriteFile(moduleRoot.getRelative("simple_rule~1.0/WORKSPACE").getPathString());
+    scratch.overwriteFile(
+        moduleRoot.getRelative("simple_rule~1.0/defs.bzl").getPathString(),
+        "def _simple_binary_impl(ctx):",
+        "  exe = ctx.actions.declare_file(ctx.label.name)",
+        "  ctx.actions.write(exe, ctx.toolchains['//:toolchain_type'].simple_info, True)",
+        "  runfiles = ctx.runfiles(files=ctx.files.data)",
+        "  for data in ctx.attr.data:",
+        "    runfiles.merge(data[DefaultInfo].default_runfiles)",
+        "  return DefaultInfo(files=depset(direct=[exe]), executable=exe, runfiles=runfiles)",
+        "simple_binary=rule(",
```

Comment:
nit: I don't think a binary that uses a toolchain is "simple", but  see where you are coming from :)

Reply:
it's now called "bare" and doesn't use a toolchain anymore. probably more apt :)